firmware/application/ui_external_items_menu_loader.cpp

```diff
@@ -0,0 +1,137 @@
+#include "ui_external_items_menu_loader.hpp"
+
+#include "sd_card.hpp"
+
+namespace ui {
+
+std::vector<Bitmap*> ExternalItemsMenuLoader::bitmaps;
+
+std::vector<GridItem> ExternalItemsMenuLoader::load_external_items(app_location_t app_location, NavigationView& nav) {
+    for (const auto& bitmap : bitmaps) {
```

Comment:
Use std::unique_ptr so we don't have deal with memory management. You can just call clear.

Reply:
awesome. I'll try that

Reply:
Does that work with that nested object?

Reply:
I added a class that wraps up all the memory stuff in the comment below if you want to use it.

Reply:
I'll leave this out as I don't know how to do this safely

Reply:
> Does that work with that nested object?

Yeah, if you use the DynamicBitmap class you’d just call clear because it cleans up its own memory.